### ISSUE TITLE
Compiled model tests failing on Windows

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#34137]
```
Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelSqlServerTest.BigModel
System.Exception : Difference found in ManyTypesEntityType.cs\r\n---- Assert.Equal() Failure: Strings differ\r\n                                  ↓ (pos 1334489)\r\nExpected: ···\"1, Uri v2) => v1 == v2,\\r\\n                \"···\r\nActual:   ···\"1, Uri v2) => v1 == null && v2 == null ||\"···\r\n                                  ↑ (pos 1334489)
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1527
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.Test[TContext](Action`1 onModelCreating, Action`1 assertModel, Func`2 useContext, Action`1 onConfiguring, CompiledModelCodeGenerationOptions options, Func`2 addServices, Func`2 addDesignTimeServices, IEnumerable`1 additionalSourceFiles, Action`1 assertAssembly, String expectedExceptionMessage, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1455
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1523
241.4330804s✘ Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelSqlServerTest.BigModel_with_JSON_columns
System.Exception : Difference found in ManyTypesEntityType.cs\r\n---- Assert.Equal() Failure: Strings differ\r\n                                  ↓ (pos 1334489)\r\nExpected: ···\"1, Uri v2) => v1 == v2,\\r\\n                \"···\r\nActual:   ···\"1, Uri v2) => v1 == null && v2 == null ||\"···\r\n                                  ↑ (pos 1334489)
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1527
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.Test[TContext](Action`1 onModelCreating, Action`1 assertModel, Func`2 useContext, Action`1 onConfiguring, CompiledModelCodeGenerationOptions options, Func`2 addServices, Func`2 addDesignTimeServices, IEnumerable`1 additionalSourceFiles, Action`1 assertAssembly, String expectedExceptionMessage, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1455
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1523
```

```
 Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelSqliteTest.BigModel
System.Exception : Difference found in ManyTypesEntityType.cs\r\n---- Assert.Equal() Failure: Strings differ\r\n                                  ↓ (pos 1229841)\r\nExpected: ···\"1, Uri v2) => v1 == v2,\\r\\n                \"···\r\nActual:   ···\"1, Uri v2) => v1 == null && v2 == null ||\"···\r\n                                  ↑ (pos 1229841)
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1527
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.Test[TContext](Action`1 onModelCreating, Action`1 assertModel, Func`2 useContext, Action`1 onConfiguring, CompiledModelCodeGenerationOptions options, Func`2 addServices, Func`2 addDesignTimeServices, IEnumerable`1 additionalSourceFiles, Action`1 assertAssembly, String expectedExceptionMessage, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1455
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1523
143.1027197s✘ Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelSqliteTest.BigModel_with_JSON_columns
System.Exception : Difference found in ManyTypesEntityType.cs\r\n---- Assert.Equal() Failure: Strings differ\r\n                                  ↓ (pos 1229841)\r\nExpected: ···\"1, Uri v2) => v1 == v2,\\r\\n                \"···\r\nActual:   ···\"1, Uri v2) => v1 == null && v2 == null ||\"···\r\n                                  ↑ (pos 1229841)
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1527
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.Test[TContext](Action`1 onModelCreating, Action`1 assertModel, Func`2 useContext, Action`1 onConfiguring, CompiledModelCodeGenerationOptions options, Func`2 addServices, Func`2 addDesignTimeServices, IEnumerable`1 additionalSourceFiles, Action`1 assertAssembly, String expectedExceptionMessage, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1455
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at Microsoft.EntityFrameworkCore.Scaffolding.CompiledModelTestBase.AssertBaseline(IReadOnlyCollection`1 scaffoldedFiles, String testName) in D:\code\efcore\test\EFCore.Specification.Tests\Scaffolding\CompiledModelTestBase.cs:line 1523
```

**Comment by ranma42** [comment]
🚀 (they also fail on Linux)

**Comment by ranma42** [comment]
#34111 is an attempt at enabling the testing of compiled models in CI to help catching these failures before merging.

**Comment by ranma42** [comment]
As mentioned in https://github.com/dotnet/efcore/pull/34138#issuecomment-2204383325 this is fixed by #34111 (it includes 6b67e70556e758720a422285f137983141f80f74 which is the same as #34138)
